cli/crates/federated-dev/src/dev/gateway_nanny.rs

```diff
@@ -57,7 +56,12 @@ pub(super) async fn new_gateway(config: Option<engine_v2::VersionedConfig>) -> O
         .into_iter()
         .map(|(k, v)| {
             (
-                k.to_string(),
+                match k {
```

Comment:
:zap: **Suggestion** 
Consider handling the case where `k` does not match any of the expected variants. This will help prevent potential runtime errors if an unexpected value is encountered.